components/create-collective/NewCreateCollective.js

```diff
@@ -0,0 +1,223 @@
+import React, { Fragment, Component } from 'react';
+import PropTypes from 'prop-types';
+import { Flex, Box } from '@rebass/grid';
+import { get } from 'lodash';
+import { defineMessages, injectIntl } from 'react-intl';
+
+import Page from '../Page';
+import { H1, P } from '../Text';
+import CreateCollectiveForm from './sections/CreateCollectiveForm';
+import CollectiveCategoryPicker from './sections/CollectiveCategoryPicker';
+import ConnectGithub from './sections/ConnectGithub';
+import SignInOrJoinFree from '../SignInOrJoinFree';
+import { withUser } from '../UserProvider';
+
+import { addCreateCollectiveMutationV2 } from '../../lib/graphql/mutations';
+import { getErrorFromGraphqlException } from '../../lib/utils';
+import { Router } from '../../server/pages';
+
+class NewCreateCollective extends Component {
+  static propTypes = {
+    host: PropTypes.object,
+    query: PropTypes.object,
+    LoggedInUser: PropTypes.object, // from withUser
+    refetchLoggedInUser: PropTypes.func.isRequired, // from withUser
+    intl: PropTypes.object.isRequired,
+    createCollectiveV2: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      collective: {},
+      result: {},
+      category: null,
+      github: null,
+      form: false,
+      error: null,
+    };
+    this.createCollective = this.createCollective.bind(this);
+    this.messages = defineMessages({
+      'host.apply.title': {
+        id: 'host.apply.title',
+        defaultMessage: 'Apply to create a new {hostname} collective',
+      },
+      'collective.create.title': {
+        id: 'collective.create.title',
+        defaultMessage: 'Create an Open Collective',
+      },
+      'collective.create.description': {
+        id: 'collective.create.description',
+        defaultMessage: 'The place for your community to collect money and share your finance in full transparency.',
+      },
+    });
+
+    this.host = props.host;
+  }
+
+  componentDidMount() {
```

Comment:
I'm sure we can find a better pattern, same for the similar section in `componentDidUpdate`.